presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -75,7 +77,13 @@
             .with(Executors.newCachedThreadPool(daemonThreadsNamed("docker-container-%d")));
 
     private final String logicalName;
-    private final Stopwatch startupTime = Stopwatch.createUnstarted();
+
+    // start is retried, we are recording the last attempt only
+    @GuardedBy("this")
+    private OptionalLong lastStartUpCommenceTimeNanos = OptionalLong.empty();
+    @GuardedBy("this")
+    private OptionalLong lastStartUFinishTimeNanos = OptionalLong.empty();
```

Comment:
This could be `Duration`

Reply:
i am storing actual data on a field and leaving rounding to the getter. I think it's a fair deal.